### PR TITLE
fix: set selected state for  `MdListOption` in selectedOptions OnInit

### LINKED ISSUE #7183: [MdListOption] selected value not checked when accessing MdSelectionList.selectedOptions.selected
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
If a MdListOption's has its ```@Input('selected')``` value set to true upon initialization, it should be included in the ```selectedOptions.selected``` attribute of its enclosing MdSelectionList.

#### What is the current behavior?
The MdListOption's initial ```@Input('selected')``` value is not respected when accessing the ```selectedOptions.selected``` attribute of its enclosing MdSelectionList.
An option that has its initial selected value to true is not included in the ```selectedOptions.selected``` but when it is subsequently click and thus gets unselected, then it is included in the ```selectedOptions.selected``` attribute of its MdSelectionList.

#### What are the steps to reproduce?

http://plnkr.co/edit/CXc5dKsoNWzU89s3endl?p=preview

Click the 'Show selected' button.
Then uncheck the selected option ('two') and click again on 'Show selected'.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular: 4.3.0
Material: 2.0.0-beta.10
Typescript: 2.3.2

### PR DESCRIPTION
**Pull request by josephperrott**
Fixes #7183

### CODE REVIEW

File: src/lib/list/selection-list.ts [hunk 1]
```diff
@@ -132,6 +132,12 @@ export class MdListOption extends _MdListOptionMixinBase
     super();
   }
 
+  ngOnInit() {
```
Comment (DevVersion) [inline src/lib/list/selection-list.ts]: Can you add the `OnInit` interface to the class?

File: src/lib/list/selection-list.spec.ts [hunk 2]
```diff
@@ -451,6 +481,13 @@ class SelectionListWithDisabledOption {
 }
 
 @Component({template: `
+<mat-selection-list>
+  <md-list-option [selected]="true">Item</md-list-option>
+</mat-selection-list>
```
Comment (DevVersion) [inline src/lib/list/selection-list.spec.ts]: The template should have a two space indentation here (as in other test components)

File: src/lib/list/selection-list.spec.ts [hunk 3]
```diff
@@ -214,6 +214,36 @@ describe('MdSelectionList', () => {
...
+    let fixture: ComponentFixture<SelectionListWithListOptions>;
+    let listItemEl: DebugElement;
+    let selectionList: DebugElement;
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [MdListModule],
+        declarations: [
+          SelectionListWithSelecedOption
+        ],
```
Comment (DevVersion) [inline src/lib/list/selection-list.spec.ts]: Nit: This can fit into a single line.

Comment (josephperrott) [reply src/lib/list/selection-list.spec.ts]: Done.

File: src/lib/list/selection-list.spec.ts (hunk 2 above)
Comment (josephperrott) [reply src/lib/list/selection-list.spec.ts]: Done

File: src/lib/list/selection-list.ts (hunk 1 above)
Comment (josephperrott) [reply src/lib/list/selection-list.ts]: Done

File: src/lib/list/selection-list.spec.ts [hunk 4]
```diff
@@ -451,6 +479,13 @@ class SelectionListWithDisabledOption {
 }
 
 @Component({template: `
+  <mat-selection-list>
+    <md-list-option [selected]="true">Item</md-list-option>
+  </mat-selection-list>`})
+class SelectionListWithSelecedOption {
```
Comment (rokikon) [inline src/lib/list/selection-list.spec.ts]: You've got a typo here SelectionListWithSelec**T**edOption

**Pull request merged**